### PR TITLE
disable ext_authz filter for http01 challenge

### LINKED ISSUE #777: Envoy ext_authz filter: `/.well-known/acme-challenge/` path is not ignored
Hello

when enabling both autoTLS and ext_authz filter, the path `/.well-known/acme-challenge/` for HTTP01 challenge is not ignored!
The traffic will necessarily pass throughout  the external authorization service (for no reason)

I think it would be great if we can add `typed_per_filter_config` to the route created for HTTP01 challenge in order to disable the filter in the route level as mentioned [here](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/ext_authz_filter#per-route-configuration), for instance

```
- name: >-
   (b2eac33b-ee65-404d-a0ab-fceaa6f26210/authztlsv2f98enkds-toto).Rules[1]
  domains:
   - authztlsv2f98enkds-toto.functions.fnc.dev.fr-par.scw.cloud
   - 'authztlsv2f98enkds-toto.functions.fnc.dev.fr-par.scw.cloud:*'
  routes:
   - match:
       prefix: >-
         /.well-known/acme-challenge/G84xbCs9GO2ENE_50aUBuOzs_T12Auy1AacLKcHcQOU
       headers:
         - name: K-Network-Hash
           exact_match: override
     route:
       weighted_clusters:
         clusters:
           - name: >-
               b2eac33b-ee65-404d-a0ab-fceaa6f26210/cm-acme-http-solver-gznmm
             weight: 100
       timeout: 0s
       upgrade_configs:
         - upgrade_type: websocket
           enabled: true
     request_headers_to_add:
       - header:
           key: K-Network-Hash
           value: >-
             55eb12aadbd590cfc4eccc0dda6c0f63df4cfac7bba366aec4224fc628813f78
         append: false
     typed_per_filter_config:
       envoy.filters.http.ext_authz:
         "@type": type.googleapis.com/envoy.extensions.filters.http.ext_authz.v3.ExtAuthzPerRoute
         disabled: true
```

### PR DESCRIPTION
**Pull request by RamyChaabane**
# Changes

:broom: Update or clean up current behavior

/kind enhancement

Fixes #777 

**Release Note**

**Docs**

### CODE REVIEW

**Comment by nak3** [comment]
/ok-to-test

Thank you so much! I will look at this PR on next Tuesday.

**Review by nak3 — COMMENTED** [review]
LGTM. Thank you @RamyChaabane 
Can we add unit test in `pkg/envoy/api/route_test.go` for `NewRouteExtAuthzDisabled`?

**Comment by RamyChaabane** [comment]
@nak3 I've added some unit tests for both `route_test.go` and `ingress_translator_test.go`

File: pkg/generator/ingress_translator_test.go [hunk 1]
```diff
@@ -719,6 +719,89 @@ func TestIngressTranslatorWithHTTPOptionDisabled(t *testing.T) {
...
+	test := struct {
+		name  string
+		in    *v1alpha1.Ingress
+		state []runtime.Object
+		want  *translatedIngress
+	}{
+		name: "http01-challenge",
+		in:   ingHTTP01Challenge("simplens", "simplename"),
+		state: []runtime.Object{
+			svcHTTPO1Challenge("simplens", "cm-acme-http-solver"),
```
Comment (nak3) [inline pkg/generator/ingress_translator_test.go:733]: `svc()` and `svcHTTPO1Challenge` do not have big diff, so we can still use `svc` with optional func like;

```suggestion
		       svc("simplens", "cm-acme-http-solver", func(svc *corev1.Service) {
			       svc.Spec.Ports = []corev1.ServicePort{{
				       Name:	   "http01-challenge",
				       TargetPort: intstr.FromInt(8089),
			       }}
		       }),
```

File: pkg/generator/ingress_translator_test.go [hunk 2]
```diff
@@ -719,6 +719,89 @@ func TestIngressTranslatorWithHTTPOptionDisabled(t *testing.T) {
...
+			epsHTTP01Challenge("simplens", "cm-acme-http-solver"),
```
Comment (nak3) [inline pkg/generator/ingress_translator_test.go:734]: Same with above `svc()`. We can override the `Subsets`.

**Review by nak3 — COMMENTED** [review]
Thank you! Leaving a comment about a few nits about util functions.

**Comment by RamyChaabane** [comment]
@nak3 maybe this is not related to this PR but Is there a reason why `config_dump` is not included in this list `'/(certs|stats(/prometheus)?|server_info|clusters|listeners|ready)?'` in the `200-bootstarp.yaml` file?
I think sometimes there is a need to access envoy config file and check how things are configured

**Comment by nak3** [comment]
Umm... `config_dump` needs to open admin interface. And so I am hesitated to open it by default because of security concern as reported in https://github.com/envoyproxy/envoy/issues/2763. Even if we use `localhost`, it still can get the vulnerability - e.g. https://github.com/projectcontour/contour/security/advisories/GHSA-5ph6-qq5x-7jwc. (Fortunately we didn't hit this vulnerability as we didn't expose the admin interface by default.) So I think we should not open the admin interface (`config_dump`) by default. 

I guess you might already know but we can access to the `config_dump` endpoint by the following steps.

#### 1. Replace admin.address with socket_address

```yaml
      admin:
        access_log_path: "/dev/stdout"
        address:
          socket_address: { address: 127.0.0.1, port_value: 9001 }
```

#### 2. Remove readiness probe from Kourier Deployment

#### 3. Port forward and acccess to the endpoint.

```shell
$ kubectl  -n kourier-system port-forward deployment/3scale-kourier-gateway 9001
```

**Comment by nak3** [comment]
/lgtm
/approve

Thank you!

**Pull request merged**